akka-stream-tests/src/test/scala/akka/stream/scaladsl/FromMaterializationSpec.scala

```diff
@@ -58,7 +61,21 @@ class FromMaterializerSpec extends StreamSpec {
         }
         .named("my-name")
 
-      source.runWith(Sink.head).futureValue shouldBe Some("setup-my-name")
+      source.runWith(Sink.head).futureValue shouldBe Some("setup-setup-my-name")
+    }
+
+    "preserve attributes of inner source" in {
+      val source = Source
+        .fromMaterializer { (_, _) =>
+          Source
+            .fromMaterializer { (_, attr) =>
+              Source.single(attr.get[MyAttribute])
+            }
+            .addAttributes(myAttributes)
+        }
+        .named("my-name")
```

Comment:
Can be skipped in the tests that does not verify name?

Reply:
Fixed.